### ISSUE TITLE
Add admin option to select which content types to apply Bagging to

### ISSUE THREAD

**Issue by mjordan** [mjordan/islandora_bagit_complex#2]
Some admins might only want to recursively Bag Books, for example. We should add an admin option to select which content types this module applies to, e.g., newspapers, newspaper issues, books, compound. Ideally, this set of options would sit within the existing Islandora Bagit admin options.

**Comment by mjordan** [comment]
As discussed in #1.

**Comment by bondjimbond** [comment]
I tested the module as-is with an update to Islandora Westvault.

In Islandora Westvault:
- Preserved a Newspaper
- Preservation datastream automatically applied to all child issues of the newspaper
- Run `drush -u 1 westvault-bagit`

Result:
- Newspaper object bag created
- Newspaper issue bags created -- twice per issue
- Newspaper page bags created -- once per page

Since I'm creating a Bag for both the Newspaper and for individual Issues, the repeating Issue bags aren't surprising -- but that last one is surprising... that all pages would be captured only once when both methods are supposed to grab everything. I'm going to do some more investigation on this.

**Comment by bondjimbond** [comment]
Test in UI, though, shows page objects getting bagged using both methods. Perhaps it's just skipping the page bags because it's already detecting them?

Once the functionality described in this issue is added, there should be no problem.

**Comment by mjordan** [comment]
Are you saying that the behavior is different if you create the bag via the UI and drush?

**Comment by mjordan** [comment]
I've added the admin option to the bottom of the Islandora BagIt admin settings form. All are off by default.

**Comment by bondjimbond** [comment]
Not exactly... The westvault-bagit drush command creates Bags for everything that's got a PRESERVATION datastream. In this case, I had the datastream on both a Newspaper object and its individual Issue objects -- so Bagit is instructed to (1) create a Bag for the Newspaper and all its children recursively, AND (2) create a Bag for each issue and its children.

The surprising outcome of this drush command is that  it created the Issue bags twice, but the Page bags only once. I would have expected it to create the Page bags twice as well.

**Comment by mjordan** [comment]
Sorry, when I said "drush command" I meant `drush --user=UID create-islandora-bag`, not `drush -u 1 westvault-bagit`. 

The Complex modules keeps track of what Bags it has created in the static array defined at https://github.com/mjordan/islandora_bagit_complex/blob/7.x/islandora_bagit_complex.module#L80. It does this so it doesn't get into an infinite loop, since the hook that fires this function is triggered by the creation of a Bag... including the current Bag. The members of this array are PIDs. If the PID is in the array, no second (or third/fourth/fifth... n) Bag should be created.

I don't know why you're seeing the surprising behavior, but I want to make sure that the Complex module is working as intended. Do you think it is?

**Comment by bondjimbond** [comment]
Yep, the Complex stuff is working just as intended. And the new menu options seem to do the trick.

**Comment by mjordan** [comment]
I'm going to close this, but open another issue if you need to.

WRT the issue bags being created twice, that sounds like unnecessary behavior - is that correct? If so, maybe you could add a static array like the one I use to check to see if a bag for the issue has already been created.

**Comment by bondjimbond** [comment]
I'm not too worried about it, since my plan is to require (or at least suggest) particular settings in Bagit if using Islandora Westvault. 

I've an issue in Islandora Westvault I'd like your eyes on if you have a moment. It should be fairly simple... will tag you there.
